py3status/modules/velib_metropole.py

```diff
@@ -0,0 +1,292 @@
+                      "[\?soft  ][\?color=deepskyblue {nb_ebike}/{nb_free_e_dock}]")
+    station_codes = [20043, 11014, 20012, 20014, 10042]
+    thresholds = {"station_state_code": [(0, "good"), (1, "bad")]}
+
+    class Meta:
+        update_config = {
+            "update_placeholder_format": [
+                {
+                    "placeholder_formats": {
+                        "density_level": ":.0f",
+                        "max_bike_overflow": ":.0f",
+                        "nb_bike": ":.0f",
+                        "nb_bike_overflow": ":.0f",
+                        "nb_dock": ":.0f",
+                        "nb_e_bike_overflow": ":.0f",
+                        "nb_e_dock": ":.0f",
+                        "nb_ebike": ":.0f",
+                        "nb_free_dock": ":.0f",
+                        "nb_free_e_dock": ":.0f",
+                        "station_gps_latitude": ":.3f",
+                        "station_gps_longitude": ".3f",
+                        "station_due_date": ".0f",
+                    },
+                    "format_strings": ["format_station"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        if not self.station_codes:
+            raise Exception("%s" % (STRING_MISSING_STATIONS))
+
+        # string to list if necessary
+        if not isinstance(self.station_codes, list):
+            self.station_codes = [self.station_codes]
+
+        self.request_timeout = 10
+
+        # take the whole map for first run
+        # default values take all stations
+        # around Paris
+        # Area is then shrink to only get
+        # specified stations
+        self.gps_top_right_latitude = 49.0
+        self.gps_top_right_longitude = 2.5
+        self.gps_bot_left_latitude = 48.6
+        self.gps_bot_left_longitude = 2.2
+        self.gps_zoom_level = 15
+
+        self.station_states = {"Operative": 0, "Close": 1}
+
+        json = self._get_velib_data()
+        if json:
+            self.number_of_stations, self.stations = self._manipulate(json)
+            self._get_usefull_area(self.stations)
+
+        # get placeholders
+        self.placeholders = []
+        for x in ["format", "format_station"]:
+            self.placeholders.append(self.py3.get_placeholders_list(x))
+
+        self.refresh = True
+        self.station_index = 1
+
+    def _camel_to_snake_case(self, data):
+        if not isinstance(data, (int, float)):
+            s1 = sub("(.)([A-Z][a-z]+)", r"\1_\2", data)
+            return sub("([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+        else:
+            return data
+
+    def _cast_number(self, value):
+        try:
+            value = float(value)
+        except ValueError:
+            try:
+                value = int(value)
+            except ValueError:
+                pass
+        return value
+
+    def _get_usefull_area(self, data):
+        """
+        reduce the zone to reduce the size of fetched data on refresh
```

Comment:
(Opinion) You're not returning anything here. Rename `_get_` to something else like `_set_`... and maybe more descriptive naming than `get_usefull_area`. Also, typo.